local-cli/runIOS/runIOS.js

```diff
@@ -64,82 +64,122 @@ function runOnDeviceByUdid(udid, scheme, xcodeProject, devices) {
 }
 
 function runOnSimulator(xcodeProject, args, inferredSchemeName, scheme){
-  try {
-    var simulators = JSON.parse(
-    child_process.execFileSync('xcrun', ['simctl', 'list', '--json', 'devices'], {encoding: 'utf8'})
-    );
-  } catch (e) {
-    throw new Error('Could not parse the simulator list output');
-  }
-
-  const selectedSimulator = findMatchingSimulator(simulators, args.simulator);
-  if (!selectedSimulator) {
-    throw new Error(`Cound't find ${args.simulator} simulator`);
-  }
-
-  const simulatorFullName = formattedDeviceName(selectedSimulator);
-  console.log(`Launching ${simulatorFullName}...`);
-  try {
-    child_process.spawnSync('xcrun', ['instruments', '-w', selectedSimulator.udid]);
-  } catch (e) {
-    // instruments always fail with 255 because it expects more arguments,
-    // but we want it to only launch the simulator
-  }
-
-  buildProject(xcodeProject, selectedSimulator.udid, scheme);
-
-  const appPath = `build/Build/Products/Debug-iphonesimulator/${inferredSchemeName}.app`;
-  console.log(`Installing ${appPath}`);
-  child_process.spawnSync('xcrun', ['simctl', 'install', 'booted', appPath], {stdio: 'inherit'});
+  return new Promise((resolve) =>
+  {
+    try {
+      var simulators = JSON.parse(
```

Comment:
Ah yes, `let simulators` outside of the try-catch block would be the preferred style.